board.h

```diff
@@ -0,0 +1,71 @@
+#include <vector>
+#include <iostream>
+#include <string>
+#include <cstdlib> 
+
+struct Position{
+    int xpos;
+    int ypos;
+
+    bool operator==(const Position b) const{
+        return ((xpos == b.xpos)
+            && (ypos == b.ypos));
+    }
+    Position(){
+
+    }
+    Position(int x, int y){
+        this->xpos = x;
+        this->ypos = y;
+    }
+};
+
+struct Mine : Position{
+    int owner;
+
+    bool operator==(const Mine b) const{
+        return ((xpos == b.xpos)
+            && (ypos == b.ypos))
+            && (owner == b.owner);
+    }
+
+    bool operator==(const Position b) const{
+        return ((xpos == b.xpos)
+            && (ypos == b.ypos));
+    }
+};
+
+struct Player{
+    int id;
+    Mine* mines;
+    int mineCount;

```

Comment:
#should initialize built-in variables (and pointers too); again, won't keep commenting to avoid spam, but apply to all occurrences 🙏.

This applies for built-in types (`int`, `char`, `bool`, etc.) and pointers (`Mine*`, `Player*`, etc.), not for objects (`std::string`, `std::vector`, etc.).